appservice/src/deploy/deployZip.ts

```diff
@@ -41,19 +42,23 @@ export async function deployZip(client: SiteClient, fsPath: string, aspPromise:
             asp = undefined;
         }
 
-        // Assume it's consumption if we can't get the plan (sometimes happens with brand new plans) and its a Function App. Consumption is recommended and more popular for functions
-        const isConsumption: boolean = (!asp && client.isFunctionApp) || Boolean(asp && asp.sku && asp.sku.tier && asp.sku.tier.toLowerCase() === 'dynamic');
-        if (client.isLinux && isConsumption) {
-            // Linux consumption doesn't support kudu zipPushDeploy
-            await deployToStorageAccount(client, zipFilePath);
-        } else {
-            await client.kudu.pushDeployment.zipPushDeploy(fs.createReadStream(zipFilePath), { isAsync: true });
-            await waitForDeploymentToComplete(client);
-            // https://github.com/Microsoft/vscode-azureappservice/issues/644
-            // This delay is a temporary stopgap that should be resolved with the new pipelines
-            await delayFirstWebAppDeploy(client, asp);
-
+        if (client.isFunctionApp && client.isLinux) {
```

Comment:
Same here, though I guess it doesn't really matter

Reply:
The other check I can get rid of, but this one I need. We validate app settings for _all_ linux function apps, regardless of if it's consumption or dedicated